pkg/limayaml/validate.go

```diff
@@ -207,7 +208,22 @@ func Validate(y *limatype.LimaYAML, warn bool) error {
 			if _, err := strconv.ParseInt(*p.Permissions, 8, 64); err != nil {
 				errs = errors.Join(errs, fmt.Errorf("field `provision[%d].permissions` must be an octal number: %w", i, err))
 			}
-		} else {
+		case limatype.ProvisionModeYQ:
+			if p.Path == nil {
+				errs = errors.Join(errs, fmt.Errorf("field `provision[%d].path` must not be empty when mode is %q", i, limatype.ProvisionModeYQ))
+				return errs
+			}
+			if !path.IsAbs(*p.Path) {
+				errs = errors.Join(errs, fmt.Errorf("field `provision[%d].path` must be an absolute path", i))
+			}
+			if p.Expression == nil {
+				errs = errors.Join(errs, fmt.Errorf("field `provision[%d].expression` must not be empty when mode is %q", i, limatype.ProvisionModeYQ))
+			}
+			// FillDefaults makes sure that p.Permissions is not nil
+			if _, err := strconv.ParseInt(*p.Permissions, 8, 64); err != nil {
+				errs = errors.Join(errs, fmt.Errorf("field `provision[%d].permissions` must be an octal number: %w", i, err))
+			}
```

Comment:
The `Path` and `Permissions` validation is duplicated, which is 75% of the code block. I wonder if they shouldn't be in a shared codepath, and only `Content`/`Expression` are separate?